sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -906,6 +912,35 @@ pi_result _pi_queue::executeCommandList(ze_command_list_handle_t ZeCommandList,
     zePrint("Command list to be executed on copy engine\n");
   auto &ZeCommandQueue =
       (UseCopyEngine) ? ZeCopyCommandQueue : ZeComputeCommandQueue;
+
+  std::unique_lock<std::mutex> ContextsLock(Device->Platform->ContextsMutex,
```

Comment:
move inside the `if` below

Reply:
I created unique_lock object here intentionally with std::defer_lock.
If I move it inside the "if" then scope of the lock is not going to be enough, its scope must be till the end of the function, otherwise new mem allocs can be created between the moment when we made a snapshot and the moment when command list is closed and executed. 
But pls notice that it is not locked because std::defer_lock is used. It is locked only if indirect access tracking enabled.
unique_lock destructor at the end of the function will unlock the mutex if it was locked (which happens only if IndirectAccessTrackingEnabled is true).
I didn't find a better to do this.

Reply:
Good point. Please put a note about needing to lock until end of the function.